modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/streaming/PagingProviderProducer.java

```diff
@@ -37,26 +39,29 @@
  */
 public final class PagingProviderProducer<T> implements Producer<List<T>> {
 
+  private static final Logger LOGGER = LoggerFactory.getLogger(PagingProviderProducer.class);
+
   public static final String COULD_NOT_OBTAIN_A_CONNECTION = "Could not obtain a connection for the configuration";
   public static final String COULD_NOT_CREATE_A_CONNECTION_SUPPLIER =
       "Could not obtain a connection supplier for the configuration";
   public static final String COULD_NOT_CLOSE_PAGING_PROVIDER = "Could not close the Paging Provider";
   public static final String COULD_NOT_EXECUTE = "Could not execute operation with connection";
   private PagingProvider<Object, T> delegate;
   private final ConfigurationInstance config;
-  private final ExtensionConnectionSupplier connectionSupplier;
+  private final ExtensionConnectionSupplier extensionConnectionSupplier;
   private final ExecutionContextAdapter executionContext;
   private final ConnectionSupplierFactory connectionSupplierFactory;
+  private ConnectionSupplier connectionSupplier;
```

Comment:
rename to `actualConnectionSupplier` or some other name that better reveals the intent